From: Daniel Borkmann <daniel@iogearbox.net>
Date: Sun, 29 Mar 2020 21:59:02 +0200
Subject: bpf, net: Fix build issue when net ns not configured
Patch-mainline: v5.7-rc1
Git-commit: 5a95cbb80ef8d8f2db29ab10777cd4742e6fc8ec
References: bsc#1177028

Fix a redefinition of 'net_gen_cookie' error that was overlooked
when net ns is not configured.

Fixes: f318903c0bf4 ("bpf: Add netns cookie and enable it for bpf cgroup hooks")
Reported-by: kbuild test robot <lkp@intel.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Gary Lin <glin@suse.com>
---
 include/net/net_namespace.h |    9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

--- a/include/net/net_namespace.h
+++ b/include/net/net_namespace.h
@@ -225,6 +225,8 @@ extern struct list_head net_namespace_li
 struct net *get_net_ns_by_pid(pid_t pid);
 struct net *get_net_ns_by_fd(int fd);
 
+u64 net_gen_cookie(struct net *net);
+
 #ifdef CONFIG_SYSCTL
 void ipx_register_sysctl(void);
 void ipx_unregister_sysctl(void);
@@ -273,8 +275,6 @@ static inline int check_net(const struct
 
 void net_drop_ns(void *);
 
-u64 net_gen_cookie(struct net *net);
-
 #else
 
 static inline struct net *get_net(struct net *net)
@@ -302,11 +302,6 @@ static inline int check_net(const struct
 	return 1;
 }
 
-static inline u64 net_gen_cookie(struct net *net)
-{
-	return 0;
-}
-
 #define net_drop_ns NULL
 #endif
 
